app/services/post_launch_job_task_service.rb

```diff
@@ -0,0 +1,39 @@
+class PostLaunchJobTaskService < TaskService
+  def process
+    create_service_instance
+    create_kafka_event
+
+    self
+  end
+
+  private
+
+  def create_service_instance
+    instance = ServiceInstance.create!(service_instance_options)
+    Rails.logger.info("ServiceInstance##{instance.id} is created.")
+  end
+
+  def create_kafka_event
+    if ENV['NO_KAFKA'].blank?
+      CatalogInventory::Api::Messaging.client.publish_topic(
+        :service => "platform.catalog-inventory.task-output-stream",
+        # TODO: what's event??
+        :event   => "Task.update",
+        :payload => kafka_payload,
+        :headers => Insights::API::Common::Request.current_forwardable
+      )
+
+      Rails.logger.info("event(Task.update) published to kafka.")
+    end
+  end
+
+  # TODO: populate input
```

Comment:
@hsong-rh Do you want to finish this since we know the attributes in the Service Instance and how they map to Tower Job Output in this case the data is coming from Task#output which will be a JSON body



Name | Description | Catalog attribute
-- | -- | --
id | The unique ID of the Job in the Ansible Tower | source_ref
name | Name of the Job | name
description | Description of the Job | description
created | The Creation Date/Time of the Credential Type in the Tower | source_created_at
status | The status of job | Stored in extra jsonb
started | The start time of the job | Stored in extra jsonb
finished | The finish time of the job | Stored in extra jsonb
artifacts | The variables exported from the job to be stored in cloud redhat.com the variables that are exported have the prefix ‘export_to_cloud_redhat_com_’ | Stored in extra jsonb
extra_vars | The input to the job, these are the variables that are sent by the catalog. It could contain passwords | Stored in extra jsonb

Reply:
The source_ref is the Tower JOB ID. You will see that in the ID field in the Task#output json